### PR TITLE
Add more gcc flags in qmstr-wrapper

### PR DESCRIPTION
**Pull request by GiasemiSh**

### CODE REVIEW

File: pkg/gccbuilder/gcc-builder.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ var (
 		"-undef":           struct{}{},
 		"-nostartfiles":    struct{}{},
 		"-remap":           struct{}{},
+		"-r":               struct{}{},
```
Comment (hemarkus) [inline pkg/gccbuilder/gcc-builder.go]: Have you checked, if this actually produces the correct results? I didn't knew about this incremental linking feature. I am worried about loosing the data that is already stored in the db.

Comment (GiasemiSh) [reply pkg/gccbuilder/gcc-builder.go]: Yes we are loosing information if we remove the -r flag. The -r generates a relocatable output file that can in turn serve as input to ld..

File: pkg/gccbuilder/gcc-builder.go [hunk 2]
```diff
@@ -174,8 +178,9 @@ func (g *GccBuilder) cleanCmdLine(args []string) {
 		}
 
 		// index string flags
-		if idx < len(args)-1 {
-			for key := range stringArgs {
+
```
Comment (hemarkus) [inline pkg/gccbuilder/gcc-builder.go]: remove this newline

File: pkg/gccbuilder/gcc-builder.go (hunk 1 above)
Comment (GiasemiSh) [reply pkg/gccbuilder/gcc-builder.go]: In the end, if we clean the -r flag we still keep the information for the relocatable output file as it's followed by linking information of the relocatable file linked to the libraries e.g. -r -o librtld.map.o dl-allobjs.os libc_pic.a

**Pull request merged**